tests/client_test.go

```diff
@@ -98,12 +108,13 @@ func TestStreamClient(t *testing.T) {
 		t.Fatalf("Error when getting servers: %v", err)
 	}
 	if !compareStreamUpstreamServers(streamServers1, streamServers) {
-		t.Errorf("Return servers %v != added servers %v", streamServers, streamServers1)
+		if len(streamServers) != 3 {
```

Comment:
I think the logic is incorrect now.
consider the case when `!compareStreamUpstreamServers(streamServers1, streamServers)` is true AND  `len(streamServers)` is 3.  In that invalid case no error will be reported.

Reply:
I don't know why this was added to be honest. Removed second `if`